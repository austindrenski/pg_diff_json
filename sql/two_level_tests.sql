-- two-level tests

\echo 'two-level test (identity)'
SELECT diff_jsonb($${
  "a": 0,
  "b": {
    "c": 2,
    "d": true
  }
}$$, $${
  "a": 0,
  "b": {
    "c": 2,
    "d": true
  }
}$$);

\echo 'two-level test (delete)'
SELECT diff_jsonb($${
  "a": 0,
  "b": {
    "c": 2,
    "d": true
  }
}$$, $${
  "a": 0,
  "b": {
    "c": 2
  }
}$$);

\echo 'two-level test (full delete)'
SELECT diff_jsonb($${
  "a": 0,
  "b": {
    "c": 2,
    "d": true
  }
}$$, $${
  "a": 0
}$$);

\echo 'two-level test (new value)'
SELECT diff_jsonb($${
  "a": 0,
  "b": {
    "c": 2,
    "d": true
  }
}$$, $${
  "a": 0,
  "b": {
    "c": 2,
    "d": false
  }
}$$);

\echo 'two-level test (new type)'
SELECT diff_jsonb($${
  "a": 0,
  "b": {
    "c": 2,
    "d": true
  }
}$$, $${
  "a": 0,
  "b": {
    "c": 2,
    "d": 3
  }
}$$);

\echo 'two-level test (new value and new type)'
SELECT diff_jsonb($${
  "a": 0,
  "b": {
    "c": 2,
    "d": true
  }
}$$, $${
  "a": 0,
  "b": {
    "c": true,
    "d": 3
  }
}$$);

\echo 'two-level test (new value and new type, multilevel)'
SELECT diff_jsonb($${
  "a": 0,
  "b": {
    "c": 2,
    "d": true
  }
}$$, $${
  "a": 1,
  "b": {
    "c": 2,
    "d": true
  }
}$$);