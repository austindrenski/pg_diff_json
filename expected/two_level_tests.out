-- two-level tests

\echo 'two-level test (identity)'
two-level test (identity)
SELECT diff_jsonb($${
  "a": 0,
  "b": {
    "c": 2,
    "d": true
  }
}$$, $${
  "a": 0,
  "b": {
    "c": 2,
    "d": true
  }
}$$);
 diff_jsonb 
------------
 
(1 row)


\echo 'two-level test (delete)'
two-level test (delete)
SELECT diff_jsonb($${
  "a": 0,
  "b": {
    "c": 2,
    "d": true
  }
}$$, $${
  "a": 0,
  "b": {
    "c": 2
  }
}$$);
     diff_jsonb     
--------------------
 {"(-,\"{b,d}\",)"}
(1 row)


\echo 'two-level test (full delete)'
two-level test (full delete)
SELECT diff_jsonb($${
  "a": 0,
  "b": {
    "c": 2,
    "d": true
  }
}$$, $${
  "a": 0
}$$);
  diff_jsonb  
--------------
 {"(-,{b},)"}
(1 row)


\echo 'two-level test (new value)'
two-level test (new value)
SELECT diff_jsonb($${
  "a": 0,
  "b": {
    "c": 2,
    "d": true
  }
}$$, $${
  "a": 0,
  "b": {
    "c": 2,
    "d": false
  }
}$$);
       diff_jsonb        
-------------------------
 {"(+,\"{b,d}\",false)"}
(1 row)


\echo 'two-level test (new type)'
two-level test (new type)
SELECT diff_jsonb($${
  "a": 0,
  "b": {
    "c": 2,
    "d": true
  }
}$$, $${
  "a": 0,
  "b": {
    "c": 2,
    "d": 3
  }
}$$);
     diff_jsonb      
---------------------
 {"(+,\"{b,d}\",3)"}
(1 row)


\echo 'two-level test (new value and new type)'
two-level test (new value and new type)
SELECT diff_jsonb($${
  "a": 0,
  "b": {
    "c": 2,
    "d": true
  }
}$$, $${
  "a": 0,
  "b": {
    "c": true,
    "d": 3
  }
}$$);
                diff_jsonb                
------------------------------------------
 {"(+,\"{b,c}\",true)","(+,\"{b,d}\",3)"}
(1 row)


\echo 'two-level test (new value and new type, multilevel)'
two-level test (new value and new type, multilevel)
SELECT diff_jsonb($${
  "a": 0,
  "b": {
    "c": 2,
    "d": true
  }
}$$, $${
  "a": 1,
  "b": {
    "c": 2,
    "d": true
  }
}$$);
  diff_jsonb   
---------------
 {"(+,{a},1)"}
(1 row)

